kalamine/layout.py

```diff
@@ -212,25 +207,25 @@ def __init__(self, filepath):
                 self.dk_index.append(dk["char"])
 
         # remove unused characters in self.dead_keys[].{base,alt}
-        def layer_has_char(char, layer_index):
-            for id in self.layers[layer_index]:
-                if self.layers[layer_index][id] == char:
-                    return True
-            return False
-
-        for dk_id in self.dead_keys:
-            base = self.dead_keys[dk_id]["base"]
-            alt = self.dead_keys[dk_id]["alt"]
+        def layer_has_char(char: str, layer_index: int) -> bool:
+            return any(
+                self.layers[layer_index][layer_id] == char
+                for layer_id in self.layers[layer_index]
+            )
+
+        for dk_dict in self.dead_keys.values():
+            base = dk_dict["base"]
+            alt = dk_dict["alt"]
             used_base = ""
             used_alt = ""
-            for i in range(len(base)):
-                if layer_has_char(base[i], Layer.BASE) or layer_has_char(
-                    base[i], Layer.SHIFT
+            for i, unknown_element in enumerate(base):
+                if layer_has_char(unknown_element, Layer.BASE) or layer_has_char(
+                    unknown_element, Layer.SHIFT
                 ):
-                    used_base += base[i]
+                    used_base += unknown_element
                     used_alt += alt[i]
-            self.dead_keys[dk_id]["base"] = used_base
-            self.dead_keys[dk_id]["alt"] = used_alt
+            dk_dict["base"] = used_base
+            dk_dict["alt"] = used_alt
```

Comment:
Bêtement, toute cette boucle `for` pète dans le refacto et va donc disparaître. Mais pour ma compréhension : pourquoi le nom `unknown_element` ? Ici l’idée était de parcourir un tableau de caractères (`base`), donc on sait assez précisément ce que l’élément contient, non ?

(Je note le coup du `enumerate`, j’adore.)